### PR TITLE
Pass tags to http backends

### PR DESCRIPTION
**Pull request by romerod**
The user_path returns tags for the user, this pull requests passes this tags to the other paths:

vhost_path
resource_path
topic_path

This makes it possible to do authorization with a tags(/roles) based approach.

See #79 for some history

### CODE REVIEW

File: src/rabbit_auth_backend_http.erl [hunk 1]
```diff
@@ -163,6 +166,9 @@ escape(K, V) ->
 
 parse_resp(Resp) -> string:to_lower(string:strip(Resp)).
 
+join_tags([]) -> "";
+join_tags(Tags) -> string:substr(lists:foldr(fun(Elem, Acc) ->  ", " ++ atom_to_list(Elem) ++ Acc end, "", Tags), 3).
```
Comment (lukebakken) [inline src/rabbit_auth_backend_http.erl]: I'd say give this a try - http://erlang.org/doc/man/string.html#join-2

**Comment by pivotal-issuemaster** [comment]
@romerod Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-auth-backend-http&pullRequestId=80)!

[Click here](https://cla.pivotal.io/sync/rabbitmq?repositoryId=rabbitmq/rabbitmq-auth-backend-http&pullRequestId=80) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: src/rabbit_auth_backend_http.erl (hunk 1 above)
Comment (michaelklishin) [reply src/rabbit_auth_backend_http.erl]: I am finishing this PR.

**Comment by pivotal-issuemaster** [comment]
@romerod Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-auth-backend-http&pullRequestId=80)!

**Comment by romerod** [comment]
As it's a list of atoms string.join didn't work without looping twice thru the list
________________________________

**Comment by michaelklishin** [comment]
Merged manually, backported to `v3.7.x`.

**Pull request closed without merging**